File: 1.fs
let lines = System.IO.File.ReadLines(".\\1.txt")
            |> List.ofSeq
            |> List.map int32

let r = lines 
        |> List.indexed
        |> List.fi(fun (idx,v) -> 
                                let found = lines.[idx+1 .. ] 
                                                |> List.tryFind (fun b -> b + v = 2020)
                                                |> Option.bind (fun b -> Some (b * v))
                                match found with
                                |> Some b -> b * v
                                |> 
                                )
        



//fuck this

File: 1.py
# %%
f = open("./1.txt","r")
# %%
vals = list(map(int, f.readlines()))
# %%
for i in range(0,len(vals)):
        for j in range(i, len(vals)):
            for k in range(j,len(vals)):
                if (vals[i] + vals[j] + vals[k] == 2020):
                    print (vals[i] * vals[j] * vals[k]) 
# %%
